File: src/2025-03-04-12-20-21--deepseek-vr-0304/javascript/exercises/practice/book-store/book-store.js
//
// This is only a SKELETON file for the 'BookStore' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const cost = (books) => {
    const counts = new Array(5).fill(0);
    for (const book of books) {
        counts[book - 1]++;
    }

    const groups = [];
    let tempCounts = [...counts];
    while (tempCounts.some(c => c > 0)) {
        tempCounts.sort((a, b) => b - a);
        let groupSize = 0;
        for (let i = 0; i < tempCounts.length; i++) {
            if (tempCounts[i] > 0) {
                groupSize++;
                tempCounts[i]--;
            }
        }
        groups.push(groupSize);
    }

    let total = 0;
    for (const size of groups) {
        switch(size) {
            case 1:
                total += 800;
                break;
            case 2:
                total += 2 * 800 * 0.95;
                break;
            case 3:
                total += 3 * 800 * 0.90;
                break;
            case 4:
                total += 4 * 800 * 0.80;
                break;
            case 5:
                total += 5 * 800 * 0.75;
                break;
            default:
                total += size * 800;
        }
    }

    const num5 = groups.filter(g => g ===5).length;
    const num3 = groups.filter(g => g ===3).length;
    const pairs = Math.min(num5, num3);
    total -= pairs * 40;

    return Math.round(total);
};

File: src/2025-03-04-12-20-21--deepseek-vr-0304/go/exercises/practice/book-store/book_store.go
package bookstore

import (
	"sort"
)

func Cost(books []int) int {
	counts := make([]int, 5)
	for _, b := range books {
		if b < 1 || b > 5 {
			continue
		}
		counts[b-1]++
	}

	sort.Sort(sort.Reverse(sort.IntSlice(counts)))

	groups := make([]int, 6)
	for {
		k := 0
		for i := 0; i < 5 && counts[i] > 0; i++ {
			k++
		}
		if k == 0 {
			break
		}
		groups[k]++

		for i := 0; i < k; i++ {
			counts[i]--
		}
		sort.Sort(sort.Reverse(sort.IntSlice(counts)))
	}

	cost := 0
	for size, num := range groups {
		switch size {
		case 5:
			cost += num * 5 * 600
		case 4:
			cost += num * 4 * 640
		case 3:
			cost += num * 3 * 720
		case 2:
			cost += num * 2 * 760
		case 1:
			cost += num * 800
		}
	}

	replacements := groups[5]
	if groups[3] < replacements {
		replacements = groups[3]
	}
	cost -= replacements * 40

	return cost
}
